### PR TITLE
OP-149 Aligned model with relational object and code cleaning/improvement

### PR DESCRIPTION
**Pull request by mwithi**
See OP-149.

### CODE REVIEW

File: src/test/java/org/isf/visits/test/TestVisit.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 	private GregorianCalendar date = new GregorianCalendar(10, 9, 8);
 	private String note = "TestNote";
 	private boolean sms = true;
-	private String duration = "10";
+	private Integer duration = new Integer(10);
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/visits/test/TestVisit.java:38]: This is unnecessary boxing; just use `private Integer duration = 10;`

File: src/test/java/org/isf/visits/test/Tests.java [hunk 2]
```diff
@@ -307,9 +293,9 @@ public void testVisitGetSet() throws Exception {
 		int id = _setupTestVisit(false);
 		Visit visit = visitManager.findVisit(id);
 
-		assertThat(visit.getDuration()).isEqualTo("10");
-		visit.setDuration("");
-		assertThat(visit.getDuration()).isEmpty();
+		assertThat(visit.getDuration()).isEqualTo(new Integer(10));
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/visits/test/Tests.java:296]: same note as above; just use `10`,

**Comment by dbmalkovsky** [comment]
Other than the two small comments, LGTM.

File: src/test/java/org/isf/visits/test/TestVisit.java (hunk 1 above)
Comment (mwithi) [reply src/test/java/org/isf/visits/test/TestVisit.java]: Right!

File: src/main/java/org/isf/visits/manager/VisitManager.java [hunk 3]
```diff
@@ -56,6 +60,38 @@
...
 	@Autowired
 	private ApplicationContext applicationContext;
+	
+	/**
+	 * Verify if the object is valid for CRUD and return a list of errors, if any
+	 *
+	 * @param visit
+	 * @throws OHServiceException
+	 */
+	private void validateVisit(Visit visit) throws OHServiceException {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/visits/manager/VisitManager.java:70]: I believe this must be made `public` as there is a call to this method in the GUI file `InsertVisit.java` at line 328

Comment (mwithi) [reply src/main/java/org/isf/visits/manager/VisitManager.java]: Yep, forgot to push, thanks!

**Pull request merged**